Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1097 @@
+                     N_ext_faces_after_eight_ways(1) <<std::endl;
+    amrex::Print()<< "Faces to be extended after eight ways extension ins z:\t" <<
+                     N_ext_faces_after_eight_ways(2) <<std::endl;
+
+    if (N_ext_faces_after_eight_ways(0) > 0) {
+        amrex::Abort("Some x faces could not be extended");
+    }
+    if (N_ext_faces_after_eight_ways(1) > 0) {
+        amrex::Abort("Some y faces could not be extended");
+    }
+    if (N_ext_faces_after_eight_ways(2) > 0) {
+        amrex::Abort("Some z faces could not be extended");
+    }
+#endif
+}
+
+/**
+* \brief this function initializes the memory for the cell FaceInfoBoxes
+*/
+void
+WarpX::InitBorrowing() {
+    int idim = 0;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][0]); mfi.isValid(); ++mfi) {
+        amrex::Box const &box = mfi.validbox();
+        auto &borrowing_x = (*m_borrowing[maxLevel()][idim])[mfi];
+        borrowing_x.inds_pointer.resize(box);
+        borrowing_x.size.resize(box);
+        borrowing_x.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_x.inds.resize(8*ncells);
+        borrowing_x.neigh_faces.resize(8*ncells);
+        borrowing_x.area.resize(8*ncells);
+    }
+
+    idim = 1;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+        amrex::Box const &box = mfi.validbox();
+        auto &borrowing_y = (*m_borrowing[maxLevel()][idim])[mfi];
+        borrowing_y.inds_pointer.resize(box);
+        borrowing_y.size.resize(box);
+        borrowing_y.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_y.inds.resize(8*ncells);
+        borrowing_y.neigh_faces.resize(8*ncells);
+        borrowing_y.area.resize(8*ncells);
+    }
+
+    idim = 2;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+        amrex::Box const &box = mfi.validbox();
+        auto &borrowing_z = (*m_borrowing[maxLevel()][idim])[mfi];
+        borrowing_z.inds_pointer.resize(box);
+        borrowing_z.size.resize(box);
+        borrowing_z.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_z.inds.resize(8*ncells);
+        borrowing_z.neigh_faces.resize(8*ncells);
+        borrowing_z.area.resize(8*ncells);
+    }
+}
+
+/**
+* \brief For the face of cell pointing in direction idim, compute the number of faces
+* we need to intrude. For the one-way extension this function returns only one or zero: one if the
+* face can be extended withe the one-way extension, zeros if it can't.
+*/
+int
+WarpX::ComputeNBorrowOneFaceExtension(amrex::Dim3 cell, amrex::Real S_ext,
+                                      const amrex::Array4<amrex::Real>& S_red,
+                                      const amrex::Array4<int>& flag_info_face,
+                                      const amrex::Array4<int>& flag_ext_face, int idim) {
```

Comment:
Can `idim` be `const`?
```suggestion
                                      const amrex::Array4<int>& flag_ext_face, const int idim) {
```

Reply:
Yes, all of them :)